FrontEnd/src/components/ProfilePage/FormComponents/FormFields/PasswordField.js

```diff
@@ -0,0 +1,65 @@
+import { useState } from 'react';
+import { PropTypes } from 'prop-types';
+import EyeInvisible from '../../../authorization/EyeInvisible';
+import EyeVisible from '../../../authorization/EyeVisible';
+import css from './PasswordField.module.css';
+
+const PasswordField = (props) => {
+    const [showPassword, setShowPassword] = useState(false);
+
+    const togglePassword = () => {
+        setShowPassword(!showPassword);
+    };
+
+    return (
+        <div className={css['password-field__item']}>
+            <div className={css['password-field__label-wrapper']}>
+                {props.requredField &&
+                <span>
+                    *
+                </span>}
+                <label
+                    htmlFor={props.inputId}
+                >
+                    {props.label}
+                </label>
+            </div>
+            <div className={css['password-field__password']}>
+                <div className={css['password-field__password__wrapper']}>
+                    <input
+                        id={props.inputId}
+                        name={props.name}
+                        type={showPassword ? 'text' : 'password'}
+                        value={props.value}
+                        onChange={props.updateHandler}
+                        placeholder={props.label}
+                        required={props.requredField}
+                    />
+                </div>
+                <span
+                    className={css['password-visibility']}
+                    onClick={togglePassword}
+                >
+                    {!showPassword ? <EyeInvisible /> : <EyeVisible />}
+                </span>
+            </div>
+            {(props.requredField || props.error) &&
```

Comment:
The error message section will display even if `props.error` is an empty string and `props.requredField` is true. It might be more logical to display this section only if there is an error message to show

Reply:
Thank you, I'll consider it.